### PR TITLE
Improve error messages when trying to use nested lists.

### PR DESCRIPTION
**Pull request by zdevito**

### CODE REVIEW

File: torch/csrc/jit/script/compiler.cpp [hunk 1]
```diff
@@ -721,6 +721,24 @@ std::shared_ptr<SugaredValue> BuiltinFunction::call(
       emitBuiltinCall(loc, *m.graph(), symbol, inputs, attributes, true));
 }
 
+inline bool isSupportedListElementType(TypePtr type) {
+  return type->isSubtypeOf(DynamicType::get()) ||
+      type->isSubtypeOf(NumberType::get());
```
Comment (apaszke) [inline torch/csrc/jit/script/compiler.cpp]: I think we also have lists of doubles

**Review by apaszke — APPROVED** [review]

File: torch/csrc/jit/script/compiler.cpp (hunk 1 above)
Comment (zdevito) [reply torch/csrc/jit/script/compiler.cpp]: It is NumberType which handles Int and Float.

Comment (apaszke) [reply torch/csrc/jit/script/compiler.cpp]: Uh of course...

**Pull request closed without merging**